e2e_playwright/st_file_uploader_test.py

```diff
@@ -0,0 +1,218 @@
+# Copyright (c) Streamlit Inc. (2018-2022) Snowflake Inc. (2022)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from playwright.sync_api import Page, expect
+
+from e2e_playwright.conftest import ImageCompareFunction, wait_for_app_run
+
+
+def test_file_uploader_render_correctly(
+    themed_app: Page, assert_snapshot: ImageCompareFunction
+):
+    """Test that the file uploader render as expected via screenshot matching."""
+    file_uploaders = themed_app.get_by_test_id("stFileUploader")
+    expect(file_uploaders).to_have_count(7)
+
+    assert_snapshot(file_uploaders.nth(0), name="st_single-file-uploader")
+    assert_snapshot(file_uploaders.nth(1), name="st_disabled-file-uploader")
+    assert_snapshot(file_uploaders.nth(2), name="st_multi-file-uploader")
+    assert_snapshot(file_uploaders.nth(4), name="st_hidden-label-file-uploader")
+    assert_snapshot(file_uploaders.nth(5), name="st_collapsed-label-file-uploader")
+
+
+def test_file_uploader_error_message_disallowed_files(
+    themed_app: Page, assert_snapshot: ImageCompareFunction
+):
+    """Test that shows error message for disallowed files."""
+    file_name1 = "example.json"
+    file_content1 = b"{}"
+
+    uploader_index = 0
+
+    with themed_app.expect_file_chooser() as fc_info:
+        themed_app.get_by_test_id("stFileUploadDropzone").nth(uploader_index).click()
+
+    file_chooser = fc_info.value
+    file_chooser.set_files(
+        files=[
+            {
+                "name": file_name1,
+                "mimeType": "application/json",
+                "buffer": file_content1,
+            }
+        ]
+    )
+
+    wait_for_app_run(themed_app)
```

Comment:
nit: Do you know if it runs into problems if we remove the:

```
wait_for_app_run(themed_app)
themed_app.wait_for_timeout(1000)
```

from the tests? I might be worth to just do one try with removing those.